### DISCUSSION TITLE
How to remove existing instances/nodes from the nats/jetstream cluster?

### DISCUSSION THREAD

**Discussion by arpanps** [nats-io/nats-server#3790] [Q&A] [answered]
In nats streaming we can remove a node by sending a NATS message to _STAN.raft.<cluster ID>.node.remove with the payload being the node ID we want to remove. like https://github.com/nats-io/nats-streaming-server/issues/1166 

how can be we do same in nats-server with jetstream?

I do see some conversation around it https://github.com/nats-io/nats-server/issues/338 however it does not provide concrete answer.

**Comment by jnmoyne** [comment, marked as answer]
You can send the process of the nats-server a "lame duck" signal telling it to shut down in an orderly manner (e.g. `nats-server --signal ldm`), but JetStream still retains the server in it's metadata (meaning it's expecting it to come back to the cluster at some point in the future), if you want to remove the server 'forever' you can then use the `nats` CLI tool to issue a 'peer remove' command for the cluster to permanently remove the server from the cluster metadata (e.g. `nats server raft peer-remove` which you will have to do using the system account).
To see what actual control message is being sent to the cluster to do that peer remove add the `--trace` option to your `nats` command.
